src/join_check/mod.rs

```diff
@@ -1,55 +1,53 @@
-use std::{sync::Arc, time::Duration};
+use std::sync::Arc;
 
-use serde::{Deserialize, Serialize};
 use teloxide::{
     prelude::*,
     types::{ChatPermissions, InlineKeyboardButton, InlineKeyboardMarkup, User},
     utils::html::escape,
 };
 
-use crate::{config::AppConfig, DialogueData, GroupDialogue, HandlerError, HandlerResult};
-pub use math_captcha::Question;
+use crate::{config::GroupsConfig, DialogueData, GroupDialogue, HandlerError, HandlerResult};
+pub use captcha::*;
 
-mod math_captcha;
-
-#[derive(Serialize, Deserialize)]
-struct CallBackData {
-    data: String,
-    btn_val: u8,
-    user_id: UserId,
-}
+mod captcha;
 
 pub async fn join_handler(
     bot: Bot,
-    app_config: Arc<AppConfig>,
+    config: Arc<GroupsConfig>,
     dialogue: GroupDialogue,
     msg: Message,
     users: Vec<User>,
 ) -> HandlerResult {
-    let Some(chat_config) = app_config.allowed_chats.iter().find(|c| c.id == msg.chat.id) else {
+    if !config.is_group_allowed(&msg.chat.id) {
         log::error!("Chat not found: {:?}", msg.chat);
```

Comment:
Let's just keep `chat.id` and `chat.name` so we can find any misuse of our but.
the whole struct is a little big.

Reply:
```suggestion
        log::error!("Unknown chat {} with id {}", msg.chat.title, msg.chat.id);
```